src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useMemo, useState } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+import { Person } from '../types/Person';
+
+type Props = {
+  peopleFromServer: Person[];
+  onSelect?: (person: Person) => void;
+};
+
+function getPreparedPeople(initialPeople: Person[], query?: string) {
+  if (!query) {
+    return initialPeople;
+  }
+
+  const preparedQuery = query.toLowerCase();
+
+  return initialPeople.filter(person => (
+    person.name
+      .toLowerCase()
+      .includes(preparedQuery)
+  ));
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  peopleFromServer,
+  onSelect = () => {},
+}) => {
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isPersonHovered, setIsPersonHovered] = useState(false);
+
+  const selectPerson = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+
+    onSelect(person);
+    setSearchQuery(person.name);
+    setIsDropdownActive(false);
+  };
+
+  const preparedPeople = useMemo(() => {
+    return getPreparedPeople(peopleFromServer, appliedQuery);
+  }, [appliedQuery, peopleFromServer]);
+
+  const applySearchQuery = useMemo(() => {
+    return debounce(setAppliedQuery, 1000);
+  }, []);
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    applySearchQuery(event.target.value);
+  };
```

Comment:
```suggestion
  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
  
    setSearchQuery(value);
    applySearchQuery(value);
  };
```
Use destructing to avoid repeating